truffle/src/main/java/org/jruby/truffle/language/constants/LookupConstantNode.java

```diff
@@ -74,9 +74,17 @@ protected RubyConstant lookupConstant(
         if (!isVisible) {
             throw new RaiseException(coreExceptions().nameErrorPrivateConstant(module, name, this));
         }
+        if(constant != null && constant.isDeprecated()){
+            warnDeprecatedConstant(name);
+        }
         return constant;
     }
 
+    @TruffleBoundary
+    private void warnDeprecatedConstant(String name) {
+        getContext().getJRubyRuntime().getWarnings().warn("constant " + name + " is deprecated");
```

Comment:
This warning does not appear to be writing to the `$stderr` that the spec is expecting.

One idea I had was to use a snippet node to call `warn()` with a warning message. Is this an okay idea? Or is there a better way to fix this?